### PR TITLE
:robot: Fix android platform detection when using p4a

### LINKED ISSUE #6851: Platform linux on some android devices
### Versions

* Kivy: master (2.0.0rc1)

### Description

Last week I had to rebuild a kivy app that was working in many android devices. I normally do it with buildozer docker image.
I was able to create the apk again and it works as expected ... in some devices! In other ones the app crashes during splash screen due to a PermissionError. 
After debugging with adb logcat I noticed kivy is not recognizing the platform correctly as it return 'linux' instead of 'android'.
This causes that [ANDROID_APP_PATH is not used](https://github.com/kivy/kivy/blob/b6eddbb2fcf97a21b7f0c4cfbf0d8ccdb74fb3ec/kivy/__init__.py#L339) and it tries to write on /data/ path.

I'm using kivy==master on buildozer.spec.

### Logs

```
04-27 14:54:56.946  7788  7807 I python  : Android kivy bootstrap done. name is main
04-27 14:54:56.946  7788  7807 I python  : AND: Ran string
04-27 14:54:56.946  7788  7807 I python  : Run user program, change dir and execute entrypoint
04-27 14:54:57.303  7788  7807 I python  :  Traceback (most recent call last):
04-27 14:54:57.303  7788  7807 I python  :    File "/home/user/hostcwd/.buildozer/android/app/main.py", line 10, in <module>
04-27 14:54:57.304  7788  7807 I python  :    File "/home/user/hostcwd/.buildozer/android/app/app.py", line 4, in <module>
04-27 14:54:57.305  7788  7807 I python  :    File "/home/use
r/hostcwd/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/myapp/kivy/__init__.py", line 351, in <module>
04-27 14:54:57.306  7788  7807 I python  :  PermissionError: [Errno 13] Permission denied: '/data/.kivy'
04-27 14:54:57.306  7788  7807 I python  : Python for android ended.
```

### PR DESCRIPTION
**Pull request by opacam**
close #6851 

**Note:**still untested

### CODE REVIEW

**Comment by AndreMiras** [comment]
Thanks for the bugfix attempt. Obviously unit tests also need to be updated.
I would say if we could narrow it down to only the check you're adding and removing the other change that introduced the bug, the code would be simpler. That's just a suggestion.
However we also need @olivier-boesch to test this method. I guess this environment variable should be always available when built with p4a (do we have other cases). Unless there's something special on his setup dropping some environment variables

**Comment by inclement** [comment]
I'd like to propose the tidied up method at https://github.com/kivy/kivy/pull/6853 - in particular it seems sensible to always allow an env var to override the detected platform, it can't hurt as a debug tool. We can also transition p4a to do this as well as setting the P4A_BOOTSTRAP variable, that would seem neater.

I made a PR just to demonstrate because I didn't want to do a big edit to @opacam's proposal, please feel free to go ahead with this one and close mine.

**Comment by olivier-boesch** [comment]
Hi everyone,

I will perform some tests later when I have time to but, making a quick verification, I've found a use case when the logic of environment cars doesn't work: if you are working with pydroid3. In this case, no env vars mentionned are set. See output below.

Maybe a mixed approach? First verify env vars and, if needed, when paths?

Output:
[INFO   ] [Logger      ] Record log in /storage/emulated/0/Android/data/ru.iiec.pydroid3/files/.kivy/logs/kivy_20-05-02_2.txt
[INFO   ] [Kivy        ] v1.11.0.dev0, git-8b6af30, 20190320
[INFO   ] [Python      ] v3.7.2 (default, Mar 20 2019, 15:02:54) 
[GCC 8.2.0]
environ({'PATH': '/data/user/0/ru.iiec.pydroid3/files:/data/user/0/ru.iiec.pydroid3/files/bin:/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/bin:/busybox-virtual:/sbin:/system/sbin:/system/bin:/system/xbin:/vendor/bin:/vendor/xbin', 'DOWNLOAD_CACHE': '/data/cache', 'ANDROID_BOOTLOGO': '1', 'ANDROID_ROOT': '/system', 'ANDROID_ASSETS': '/system/app', 'ANDROID_DATA': '/data', 'ANDROID_STORAGE': '/storage', 'EXTERNAL_STORAGE': '/sdcard', 'ASEC_MOUNTPOINT': '/mnt/asec', 'BOOTCLASSPATH': '/system/framework/com.qualcomm.qti.camera.jar:/system/framework/QPerformance.jar:/system/framework/core-oj.jar:/system/framework/core-libart.jar:/system/framework/conscrypt.jar:/system/framework/okhttp.jar:/system/framework/bouncycastle.jar:/system/framework/apache-xml.jar:/system/framework/legacy-test.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/ims-common.jar:/system/framework/org.apache.http.legacy.boot.jar:/system/framework/android.hidl.base-V1.0-java.jar:/system/framework/android.hidl.manager-V1.0-java.jar:/system/framework/tcmiface.jar:/system/framework/WfdCommon.jar:/system/framework/oem-services.jar:/system/framework/qcom.fmradio.jar:/system/framework/telephony-ext.jar', 'SYSTEMSERVERCLASSPATH': '/system/framework/services.jar:/system/framework/ethernet-service.jar:/system/framework/wifi-service.jar:/system/framework/com.android.location.provider.jar', 'ANDROID_SOCKET_zygote': '9', 'PKG_CONFIG_PATH': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/pkgconfig', 'SSL_CERT_FILE': '/data/user/0/ru.iiec.pydroid3/files/cert.pem', 'TERMINFO': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/share/terminfo', 'LD_LIBRARY_PATH': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib', 'TMPDIR': '/data/user/0/ru.iiec.pydroid3/cache', 'SHELL': '/data/user/0/ru.iiec.pydroid3/files/sh', 'CONFIG_SHELL': '/data/user/0/ru.iiec.pydroid3/files/sh', 'LIBRARY_PATH': '/data/user/0/ru.iiec.pydroid3/files/lib/gcc/aarch64-linux-android/8.2.0/', 'XDG_CACHE_HOME': '/data/user/0/ru.iiec.pydroid3/cache', 'APP_PACKAGE': '/data/app/ru.iiec.pydroid3-EdWP4V2iZJ6TydGHILDxKQ``/base.apk', 'LC_ALL': 'en_US.UTF-8', 'JOBLIB_MULTIPROCESSING': '0', 'KERAS_BACKEND': 'theano', 'PYDROID3_LICENSE': 'Accepted', 'PYDROID_RPC': '/data/user/0/ru.iiec.pydroid3/files/RPCServer.socket', 'PYDROID_PACKAGE': 'ru.iiec.pydroid3', 'BROWSER': 'mybrowser %s', 'ANDROID_PRIVATE': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android', 'ANDROID_ARGUMENT': '/storage/emulated/0/Android/data/ru.iiec.pydroid3/files', 'HOME': '/data/user/0/ru.iiec.pydroid3/app_HOME', 'PYDR_STDERROUT': '/data/user/0/ru.iiec.pydroid3/files/pydroid_stderrout_log', 'ANDROID_APP_PATH': '/storage/emulated/0/Android/data/ru.iiec.pydroid3/files', 'ANDROID_ORIGFNAME': '/data/user/0/ru.iiec.pydroid3/files/temp_iiec_codefile.py', 'ANDROID_ENTRYPOINT': '/data/user/0/ru.iiec.pydroid3/files/temp_iiec_codefile.py', 'PYTHONHOME': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android', 'PYTHONPATH': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android:/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib', 'PYSDL2_DLL_PATH': '/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib', 'PYTHON_NAME': 'python'})

**Comment by inclement** [comment]
In this case the `ANDROID_ARGUMENT` env var that was originally removed from kivy's check is present - which makes sense because pydroid3 is ensuring kivy will work.

We should definitely leave the ANDROID_ARGUMENT check in for backwards compatibility, with a comment about the fact that it used to be what we reliably used. @opacam could you add that back in as an alternative to P4A_BOOTSTRAP being present?

I'm not against adding other checks for being in an android environment, if necessary. Perhaps we should be using pyjnius to query the android api.

**Comment by opacam** [comment]
> We should definitely leave the ANDROID_ARGUMENT check in for backwards compatibility, with a comment about the fact that it used to be what we reliably used. @opacam could you add that back in as an alternative to P4A_BOOTSTRAP being present?

Yes, no problem

File: kivy/utils.py [hunk 1]
```diff
@@ -424,15 +424,17 @@ def format_bytes_to_human(size, precision=2):
...
-    # existence of system specific commands and files
-    if (
-            path.exists('/default.prop')
-            and path.exists('/system/bin/logcat')
-            and path.exists('/system/xbin')
-    ):
+    # existence of environ variables set during Python initialization
+    kivy_build = environ.get('KIVY_BUILD', '')
+    if kivy_build in {'android', 'ios'}:
+        return environ['KIVY_BUILD']
```
Comment (AndreMiras) [inline kivy/utils.py:430]: minor: I was just wondering why not using the `kivy_build` variable since it was created already
```
return kivy_build
```

Comment (opacam) [reply kivy/utils.py:430]: Yes, I forgot to change that, I will adres this in another PR, thanks @AndreMiras!!

Comment (AndreMiras) [reply kivy/utils.py:430]: I was being curious, but I don't think it matter that much.
Thanks again for the quick fix

Comment (opacam) [reply kivy/utils.py:430]: @AndreMiras, 

Done at #6854

Thanks again!!

**Pull request merged**